riscv-rt/src/interrupts.rs

```diff
@@ -75,29 +75,4 @@ pub unsafe extern "C" fn _dispatch_core_interrupt(code: usize) {
     any(target_arch = "riscv32", target_arch = "riscv64"),
     feature = "v-trap"
 ))]
-core::arch::global_asm!(
-    r#" .section .trap, "ax"
-        .weak _vector_table
-        .type _vector_table, @function
-        
-        .option push
-        .balign 0x4 // TODO check if this is the correct alignment
-        .option norelax
-        .option norvc
-        
-        _vector_table:
-            j _start_trap                     // Interrupt 0 is used for exceptions
-            j _start_SupervisorSoft_trap
-            j _start_DefaultHandler_trap      // Interrupt 2 is reserved
-            j _start_MachineSoft_trap
-            j _start_DefaultHandler_trap      // Interrupt 4 is reserved
-            j _start_SupervisorTimer_trap
-            j _start_DefaultHandler_trap      // Interrupt 6 is reserved
-            j _start_MachineTimer_trap
-            j _start_DefaultHandler_trap      // Interrupt 8 is reserved
-            j _start_SupervisorExternal_trap
-            j _start_DefaultHandler_trap      // Interrupt 10 is reserved
-            j _start_MachineExternal_trap
-        
-        .option pop"#
-);
+riscv_macros::vector_table!();
```

Comment:
```suggestion
riscv::vector_table!();
```

In fact, `riscv-rt/src/interrupts.rs` can just simply use the `#[riscv::pac_enum(unsafe CoreInterruptNumber)]` macro over an enumeration. In that case, the `vector_table` macro is not necessary.

Reply:
I was thinking about this, but there's the `weak`/`global` symbol issue. We would need to control this from an attribute.

Reply:
That said, I can't find if we guarantee that `_vector_table` is weak in `riscv-rt`, so we might be able to make it strong instead. And then we can use `pac_enum`.

Reply:
In this case, I'd leave it as a strong symbol. PACs should ensure that the table fits the target specifications and can enable the `riscv-rt/no-interrupts` feature to avoid `riscv-rt` from generating the table when needed.